File: C#/nofiveweekends.cs
using System;

public class Test {
    
    //Method to determine which day of the week the next year starts on,
    //given the first day of the current year and whether or not it is a leap year
    public static int nextYearFirstDay(int currentYearFirstDay, bool isLeapYear){
        if (isLeapYear){
           return (366 + currentYearFirstDay) % 7;
        }
        
        else{
            return (365 + currentYearFirstDay) % 7;
        } 
    }
    
    //Method to determine if a year does not have a month with 5
    //weekends. his only happens on leap years that start on a
    //Sunday and normal years that start on a Monday
    public static bool hasNoFiveWeekends(bool isLeapYear, int firstDayOfYear){
        
        if(firstDayOfYear == 0 && isLeapYear){
            return true;   
        }
        else if(firstDayOfYear == 1 && !isLeapYear){
            return true;
        }
        else{
            return false;
        }
    }
        
    //Declare the year range constants
    public const int startYear = 2000;
    public const int endYear = 2100;
   
    //Sunday is 0, Monday is 1, Tuesday is 2, etc.
    public static int firstDayOfYear = 6; //Jan 1, 2000 was a Saturday
    public static bool isLeapYear;
    
    public static void Main(string[] args){
        
        //Loop through the specified year range
        for(int i = startYear; i <= endYear; i++){
            
            //Check if it is a leap year
            isLeapYear = false;
            if(i%4 == 0){
                isLeapYear = true;
            }
            
            //Print out the non-five weekend years
            if(hasNoFiveWeekends(isLeapYear, firstDayOfYear)){
                Console.Write(i + ", ");
            }
                
            firstDayOfYear = nextYearFirstDay(firstDayOfYear, isLeapYear);
        }
    }
}

File: Clojure/nofiveweekends.clj
(ns nofiveweekends)

;;Method to determine which day of the week the next year
;;starts on given the first day of the current year and
;;whether or not it is a leap year
(defn nextYearFirstDay [currentDayFirstYear, isLeapYear] (

   if (= isLeapYear 1)

     (do ;;366 days if leap year
        (def firstDay (mod (+ currentDayFirstYear 366) 7)))

     (do ;;365 days if normal year
       (def firstDay (mod (+ currentDayFirstYear 365) 7)))))

;;Method to determine if a year does not have a month
;;with 5 weekends. This only happens on leap years that
;;start on a Sunday and normal years that start on Monday
(defn hasNoFiveWeekends [isLeapYear, firstDayOfYear] (
                                                       
   if (or (and (= firstDayOfYear 0) (= isLeapYear 1))
          (and (= firstDayOfYear 1) (= isLeapYear 0)))
                                                       
     (do
        (def noFiveWeekends 1))
                                                       
     (do
        (def noFiveWeekends 0))))


;;declare the year range constants
(def startYear 2000)
(def endYear 2100)

;;Sunday is 0, Monday is 1, Tuesday is 2, etc.
(def firstDayOfYear 6);;Jan 1, 2000 was a Saturday
(def isLeapYear)


;; loop through the specified year range
(loop [startYear 2000]
  (when (<= startYear endYear)

  (def isLeapYear)

  ;;Check if current year is a leap year
  (if (= (mod startYear 4) 0)

    (do
      (def isLeapYear 1))

    (do
      (def isLeapYear 0)))
    

  (def noFiveWeekends 2)
  
  (hasNoFiveWeekends isLeapYear firstDayOfYear)

  ;;Print all non-five weekend years
  (if (= noFiveWeekends 1)

    (do
      (print startYear ", ")))

 (nextYearFirstDay firstDayOfYear isLeapYear)
 (def firstDayOfYear firstDay)
 
 (recur (+ startYear 1))))
